### PR TITLE
feat(bounty): add report button that append data to a spreadsheet using Rows API

### PR DESCRIPTION
**Pull request by crocodile-bounty-hunter**
### Overview

This PR introduces a significant enhancement to the Help state by integrating user feedback and error reporting with Rows spreadsheet submissions, replacing the previous "Talk to Us" feature. It also automates the setup of necessary configurations using GitHub Actions.

### Key Changes
**1. Action Replacement and Reporting:**
- The "Talk to Us" feature is replaced by an automated submission process to a Rows spreadsheet.
- For pages without detectable tables, a "Report" button logs details (UTC datetime, URL, domain, browser details, and report type) to the spreadsheet.
- A "Report" link is persistently displayed on all pages, allowing users to report issues related to table detection or other concerns via a modal with multiple choices.
**2. Configuration and Automation:**
- All necessary Rows API keys are now included in the environment configuration file.
- A new GitHub Action step has been added to automatically generate this configuration file during the release process, ensuring that all settings are correctly applied without manual intervention.

This update aims to streamline user feedback mechanisms and enhance data capture accuracy for better service improvements.

### CODE REVIEW

File: src/utils/rows-api/report.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  const row_cells = [
+    new Date().toUTCString(),
+    tab.url,
+    new URL(tab.url!).hostname,
+    navigator.userAgent,
+    navigator.userAgent,
+    'no table detected'
+  ]
+
+  await fetch.post(`https://api.rows.com/v1/spreadsheets/${SHEET_ID}/tables/${TABLE_ID}/values/A1:F99999:append`, {
```
Comment (ricardofelgueiras) [inline src/utils/rows-api/report.ts:42]: You can replace `A1:F99999` with `A1:F` 💅

File: src/utils/rows-api/report.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  'Domain',
+  'Browser',
+  'Browser version',
+  'Report type'
+];
+
+const SHEET_ID = '1aL6dsP2jqENGlkSq3A5O9';
+const TABLE_ID = '87fa52ac-40fc-4d3f-9aff-0cb98f0e9b0d';
+
+export async function createReportHeader() {
```
Comment (ricardofelgueiras) [inline src/utils/rows-api/report.ts:16]: IMHO, you can leave this out and assume that the spreadsheet is already created and it contains the headers as well.

File: src/utils/rows-api/report.ts (hunk 1 above)
Comment (victorfern91) [reply src/utils/rows-api/report.ts:42]: Nice, thanks!

File: src/utils/rows-api/report.ts (hunk 2 above)
Comment (victorfern91) [reply src/utils/rows-api/report.ts:16]: Ok, that makes sense, I will remove it.

**Comment by ricardofelgueiras** [comment]
image

image

**Comment by ricardofelgueiras** [comment]
image
image

**Comment by crocodile-bounty-hunter** [comment]
Hey @ricardofelgueiras thanks for you feedback. Regarding the "feedback form". 

What do you think about this?

image

**Comment by ricardofelgueiras** [comment]
> Hey @ricardofelgueiras thanks for you feedback. Regarding the "feedback form".
> 
> What do you think about this?
> 
> image

It's ok.

**Comment by ricardofelgueiras** [comment]
A few comments on the PR:
- Add the `.env` file to the `.gitignore`
- Change the `deploy` action on the step `Build chrome extension dist files` and start passing the three env variables via secret. On my side I'll make sure the variables are configured on the repository before you merge it.
- Add a mention to the .env file on the README file explaining why and for what's is being used.

**Comment by crocodile-bounty-hunter** [comment]
> A few comments on the PR:
> 
> * Add the `.env` file to the `.gitignore`
> * Change the `deploy` action on the step `Build chrome extension dist files` and start passing the three env variables via secret. On my side I'll make sure the variables are configured on the repository before you merge it.
> * Add a mention to the .env file on the README file explaining why and for what's is being used, and how to set it up when running the extension locally.
> 
> Thank you.

Done!

File: src/utils/rows-api/fetch.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      Authorization: `Bearer ${import.meta.env.VITE_ROWS_API_KEY}`,
+    },
+    body: method !== 'GET' ? JSON.stringify(data) : null,
+  });
+
+  return response.json();
+}
+
+export default {
+  get: function (url: string) {
```
Comment (ricardofelgueiras) [inline src/utils/rows-api/fetch.ts:18]: Unused code. Please delete it.

**Review by ricardofelgueiras — APPROVED** [review]

**Review by crocodile-bounty-hunter — COMMENTED** [review]
solve comments from @ricardofelgueiras

**Pull request merged**